tests/test_vision_tagger_telemetry.py

```
@@ -0,0 +1,140 @@
+            with pytest.raises(RuntimeError):
+                tag_image("https://example.com/a.jpg")
+        payload = _parse_log(caplog)
+        assert payload["status"] == "failure"
+        assert payload["error"] == "schema_invalid"
+
+    def test_fetch_timeout(self, caplog):
+        caplog.set_level(logging.WARNING, logger=LOGGER_NAME)
+        with patch.object(
+            vision_tagger, "_fetch_image", side_effect=httpx.ReadTimeout("slow")
+        ):
+            with pytest.raises(httpx.TimeoutException):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)